orchagent/switchorch.cpp

```diff
@@ -472,24 +478,230 @@ void SwitchOrch::doAppSwitchTableTask(Consumer &consumer)
     }
 }
 
-void SwitchOrch::doTask(Consumer &consumer)
+bool SwitchOrch::setSwitchHashFieldListSai(sai_object_id_t oid, std::vector<sai_int32_t> &hfList) const
+{
+    sai_attribute_t attr;
+
+    attr.id = SAI_HASH_ATTR_NATIVE_HASH_FIELD_LIST;
+    attr.value.s32list.list = hfList.data();
+    attr.value.s32list.count = static_cast<sai_uint32_t>(hfList.size());
+
+    auto status = sai_hash_api->set_hash_attribute(oid, &attr);
+    return status == SAI_STATUS_SUCCESS;
+}
+
+bool SwitchOrch::setSwitchHashEcmpHash(const SwitchHash &hash) const
```

Comment:
Please combine this two functions to avoid repeating the code. Pass bool for ecmp vs lag

Reply:
@prsunny done